presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java

```diff
@@ -148,6 +149,37 @@ public void testTableWithNoSupportedColumns()
         }
     }
 
+    @Test
+    public void testInsertWithFailureDoesntLeaveBehindOrphanedTable()
+            throws Exception
+    {
+        String schemaName = format("tmp_schema_%s", UUID.randomUUID().toString().replaceAll("-", ""));
+        try (
+                AutoCloseable schema = withSchema(schemaName);
+                AutoCloseable table = withTable(format("%s.test_cleanup", schemaName), "(x INTEGER)")) {
+            assertQuery(format("SELECT table_name FROM information_schema.tables WHERE table_schema = '%s'", schemaName), "VALUES 'test_cleanup'");
+
+            execute(format("ALTER TABLE %s.test_cleanup ADD CHECK (x > 0)", schemaName));
+
+            assertQueryFails(format("INSERT INTO %s.test_cleanup (x) VALUES (0)", schemaName), "ERROR: new row .* violates check constraint [\\s\\S]*");
```

Comment:
Why not use `.*` at the end?

Reply:
Unfortunately `.*` won't match newlines

Reply:
- i think it makes sense for `com.facebook.presto.tests.QueryAssertions#assertExceptionMessage` to compile pattern with `DOTALL` flag
- you can embed DOTALL flag in the pattern -- `"(?s)ERROR: ... .*"` -- which is arguably more readable, as it directly conveys the intent